### PR TITLE
Added an example of multi worker training with TF 2

### PR DESCRIPTION
**Pull request by zaccharieramzi**

### CODE REVIEW

**Comment by lesteve** [comment]
I changed the doc so that the example can be found from the doc. I am going to merge it like this. If you want to improve the example README (for example say in a few words why it is simpler than using Horovod), be my guest!

Also a question for @zaccharieramzi and @remicres, why `SBATCH -A` in the Slurm submission script? Is this only useful if you have multiple projects?

**Comment by mdiazmel** [comment]
> Also a question for @zaccharieramzi and @remicres, why `SBATCH -A` in the Slurm submission script? Is this only useful if you have multiple projects?

Hi @lesteve,

`#SBATCH -A` in the header of a slurm script stands for `--account`. It's only useful if you have multiple projects in Jean-Zay (to be sure that the hours will be charged to the right project).

**Comment by lesteve** [comment]
OK this is what I thought. I think we should remove this from our doc for simplicity. My feeling is that:
- most people have a single Jean-Zay project
- the few people that have multiple projects very likely know about this quirk already

**Comment by remicres** [comment]
Yes we can remove that, it's only used to select the account on which it runs

File: docs/examples/tf/tf_distributed/mnist_example.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    # limit imports oustide the call to the function, in order to launch quickly
+    # when using dask
+    import tensorflow as tf
+    from tensorflow import keras
+    from tensorflow.keras import layers
+    # model building
+    tf.keras.backend.clear_session()  # For easy reset of notebook state.
+
+    slurm_resolver = tf.distribute.cluster_resolver.SlurmClusterResolver(port_base=15000)
+    mirrored_strategy = tf.distribute.MultiWorkerMirroredStrategy(cluster_resolver=slurm_resolver)
```
Comment (remicres) [inline docs/examples/tf/tf_distributed/mnist_example.py:20]: As mentioned @mypey on gitter we should employ `communication=tf.distribute.experimental.CollectiveCommunication.NCCL` here

Comment (zaccharieramzi) [reply docs/examples/tf/tf_distributed/mnist_example.py:20]: yes that's totally true, I didn't think of changing it in the PR before it was merged. We can have a small PR to fix this.

Comment (remicres) [reply docs/examples/tf/tf_distributed/mnist_example.py:20]: #59

**Pull request merged**